File: frontend/src/types/creatorMilestone.ts
export interface CreatorMilestone {
  id: number
  type: 'SUBSCRIBERS' | 'VIEWS' | 'VIDEOS' | 'WATCH_HOURS' | 'REVENUE' | 'CUSTOM'
  title: string
  description: string
  targetValue: number
  currentValue: number
  progress: number
  platform: string
  status: 'PENDING' | 'IN_PROGRESS' | 'ACHIEVED' | 'EXPIRED'
  achievedAt?: string
  targetDate?: string
  createdAt: string
}

export interface MilestoneTimeline {
  id: number
  milestoneId: number
  date: string
  value: number
  note?: string
}

export interface MilestoneReward {
  id: number
  milestoneId: number
  type: 'BADGE' | 'FEATURE_UNLOCK' | 'CREDITS' | 'CUSTOM'
  title: string
  description: string
  isRedeemed: boolean
}

export interface CreatorMilestoneSummary {
  totalMilestones: number
  achievedCount: number
  inProgressCount: number
  nextMilestone: string
  achievementRate: number
}

export interface CreateMilestoneRequest {
  type: string
  title: string
  description: string
  targetValue: number
  platform: string
  targetDate?: string
}

File: backend/onGo-domain/src/main/kotlin/com/ongo/domain/creatormilestone/CreatorMilestone.kt
package com.ongo.domain.creatormilestone

import java.math.BigDecimal
import java.time.LocalDate
import java.time.LocalDateTime

data class CreatorMilestone(
    val id: Long = 0,
    val workspaceId: Long,
    val type: String,
    val title: String,
    val description: String,
    val targetValue: Long,
    val currentValue: Long = 0,
    val progress: BigDecimal = BigDecimal.ZERO,
    val platform: String,
    val status: String = "PENDING",
    val achievedAt: LocalDateTime? = null,
    val targetDate: LocalDate? = null,
    val createdAt: LocalDateTime = LocalDateTime.now(),
    val updatedAt: LocalDateTime = LocalDateTime.now(),
)
